thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/email/filter/BaseAlertFilter.java

```diff
@@ -50,7 +71,12 @@ public void setParameters(Map<String, String> parameterSetting) {
         Field field = c.getDeclaredField(fieldName);
         boolean accessible = field.isAccessible();
         field.setAccessible(true);
-        field.set(this, Double.valueOf(value));
+        if (isNumeric) {
```

Comment:
This should be the other way around. Depend on the declared field type, not the input data - i.e. if the field has a numeric type, attempt conversion.